### ISSUE TITLE
[AUTOSRE] fix(bot-ta-analysis): Validation failed for ichimoku_cloud_momentum: risk parameters must be positive. SL: 0....

### ISSUE THREAD

**Issue by yurisa2** [PetroSa2/petrosa-bot-ta-analysis#320]
# Problem Statement

The `ichimoku_cloud_momentum` strategy emits signals with `stop_loss = 0.0` and
`take_profit = 0.0` when it receives degenerate (zero-price) candles. The signal engine correctly
rejects them:

```
ERROR - ta_bot.core.signal_engine - Validation failed for ichimoku_cloud_momentum:
  risk parameters must be positive. SL: 0.0, TP: 0.0
```

At filing (2026-09-23T06:04Z) this was firing 272 times/hour on one pod and 206 on another
(3.5% and 2.7% of all log lines). The result is **total, silent signal loss for this strategy** —
every signal it produces is discarded.

**The symptom is currently dormant, and the defect is still present on `main`.** Verified
2026-09-23: 0 occurrences across the full life of both current pods. A deploy did land in between
(`6ec5c37`, "fix(#303): raise candle warm-up to 400"), but **it did not touch ichimoku** —
`git log --oneline origin/main -- ta_bot/strategies/ichimoku_cloud_momentum.py` returns exactly one
commit, `0a046c1` (2026-09-20, a NATS change). No SL/TP guard has ever been added to this strategy.
The corrupt input stopped; the code did not change. It will recur.

Reproduced deterministically against the deployed code:

```
downtrend + 26 zero-price bars: action=sell SL=0.0  TP=0.0      price=0.0
downtrend +  5 zero-price bars: action=sell SL=52.18 TP=-104.36 price=0.0
```

# Scope

Add the repo's existing degenerate-data guard to one strategy, plus a deterministic regression test.

**Root cause.** Candles with `high`/`low`/`close` == 0 reach an unguarded strategy:

- `ta_bot/strategies/ichimoku_cloud_momentum.py:137` — `kijun_sen = (high_26 + low_26) / 2`.
  With 26+ consecutive zero bars, `kijun_sen == 0.0`.
- `:89` and `:94` — `stop_loss = current_ichimoku["kijun_sen"]` → `0.0`
- `:90`/`:95` — `risk = abs(current["close"] - stop_loss)` → `0.0` when `close == 0.0`
- `:91`/`:96` — `take_profit = close ± risk * 2` → `0.0`

Why a signal is emitted at all: `senkou_span_a`/`senkou_span_b` are `.shift(26)` (`:140`, `:145`),
so the cloud still carries valid **pre-corruption** prices. `_check_bearish_ichimoku` (`:191-202`)
then evaluates `0.0 < senkou_a and 0.0 < senkou_b` → True, and `cloud_bearish` is True in a falling
market. This is also why fewer than 26 zero bars yields a *negative* take-profit instead of zero,
and why an all-zero frame yields no signal at all.

**This is NOT candle starvation.** Ichimoku requires 52 bars
(`ta_bot/strategies/ichimoku_cloud_momentum.py:31`, `if len(df) < 52`); the fetch limit is 400
(`ta_bot/services/data_manager_client.py:31`, `MIN_WARMUP_CANDLES = 400`, and 250 before `6ec5c37`).
78 < 250 < 400 — it was never starved. Starvation produces **NaN** spans, and NaN comparisons are
False, so it yields *no signal*, never `SL: 0.0`. So this is not a duplicate of #303 (closed) nor of
`petrosa_k8s#1112`'s warm-up half; it belongs to that epic's *other* half, "consumers fail on
degenerate data" — the same family as #307/#309 and #316/#317, which were fixed per-strategy and
skipped ichimoku.

**Out of scope — do not touch:**

- `ta_bot/core/signal_engine.py:524-534` (message at `:531`) and the duplicate at
  `ta_bot/models/signal.py:219-225`. The validator is correct and is the last line of defence.
  Relaxing it to accept zero would convert a caught error into a zero-stop-loss order.
- `MIN_WARMUP_CANDLES`. Already 400; raising it changes nothing here.
- Any central/shared sanitizer. The repo's established convention is a per-strategy guard.
- The two unrelated live defects found alongside this one (doji_reversal `SignalStrength.LOW`
  `AttributeError`; `persist_signals_batch` silent MySQL failure). Separate tickets — folding them
  in will blow the one-run budget.

# Acceptance Criteria

- [ ] `ta_bot/strategies/ichimoku_cloud_momentum.py` imports `math` and, in `analyze()` immediately
      after `current = self._get_current_values(...)` (line 41), returns `None` when the close is
      not usable:
      `if not math.isfinite(current["close"]) or current["close"] <= 0: return None`
- [ ] In the same function, after `stop_loss` and `take_profit` are computed (lines 89-96) and
      **before** the `Signal` is constructed, both are guarded:
      `if not math.isfinite(stop_loss) or stop_loss <= 0: return None` and the same for
      `take_profit`. The close-guard alone does NOT catch the negative-take-profit variant
      reproduced above, so both guards are required.
- [ ] The guard mirrors the already-merged pattern in
      `ta_bot/strategies/rsi_extreme_reversal.py:122-128` (whose comment documents this exact
      failure mode) and `ta_bot/strategies/ema_alignment_bearish.py:84`. Copy that shape rather than
      inventing a new one.
- [ ] `tests/test_risk_validation.py` — extend
      `test_corrupted_hourly_sweep_skips_affected_strategies_cleanly` (line 152). It currently
      zeroes only the **final** `close` and passes
      `enabled_strategies=["ema_alignment_bearish", "rsi_extreme_reversal"]`, so ichimoku is never
      exercised. It must:
  - build a >=130-bar strictly descending OHLCV frame, then set `open/high/low/close = 0.0` on the
    **last 26 rows** (26 is required — `kijun_sen` is a 26-period window; one zero bar produces
    `SL=52.18, TP=-104.36`, not `SL: 0.0, TP: 0.0`)
  - add `"ichimoku_cloud_momentum"` to `enabled_strategies`
  - assert `signals == []` **and** that no caplog record contains `"risk parameters must be positive"`
  - add a second case zeroing only the last 5 rows, asserting no signal with a negative
    `take_profit` escapes
- [ ] `tests/strategies/test_ichimoku_cloud_momentum.py` gains a direct unit case asserting
      `strategy.analyze(corrupt_df, meta) is None` for the 26-zero-bar frame. (This file has no
      degenerate-data case today.)
- [ ] `make pipeline` passes in `petrosa-bot-ta-analysis`.

# Technical Context

**Repository:** `PetroSa2/petrosa-bot-ta-analysis` (issue repo and implementation repo are the same).

**Files to change:**

| File | Change |
|---|---|
| `ta_bot/strategies/ichimoku_cloud_momentum.py` | `import math` + two guards in `analyze()` |
| `tests/test_risk_validation.py` (test at line 152) | extend with ichimoku + 26 zero bars, plus 5-bar case |
| `tests/strategies/test_ichimoku_cloud_momentum.py` | direct degenerate-data unit case |

**Files to read for the pattern (do not edit):**

- `ta_bot/strategies/rsi_extreme_reversal.py:122-128` — the canonical guard, with a comment naming this exact failure
- `ta_bot/strategies/ema_alignment_bearish.py:84` — same guard, second precedent
- `ta_bot/core/signal_engine.py:524-534` — the validator that rejects the bad signal

**Branch from `origin/main`.** Note `.serena/project.yml` shows as modified in the working tree; that
is pre-existing and unrelated — do not commit it.

**Deliberately excluded from the ACs.** The auto-filed criterion *"Service operates across
consecutive hourly sweeps with zero unhandled exceptions"* cannot be verified inside a CI run: it
needs two hours of production observation, and the trigger is market-data dependent and currently
absent. The deterministic unit test replaces it. Do not attempt to reproduce this in production.

**Related:**

- #303 (CLOSED) — candle warm-up raised to 400. Different root cause; not a duplicate.
- #307/#309, #316/#317 — the same degenerate-data guard applied to other strategies.
- `PetroSa2/petrosa_k8s#1112` — "Candle sufficiency contract" epic. This ticket is its
  *"consumers fail on degenerate data"* half, not its warm-up half.

# Confidence Report

- **Score:** 0.93 (threshold 0.80)
- **Problem statement (0.30):** full — failure reproduced deterministically against the deployed
  code, with exact SL/TP values for both the 26-bar and 5-bar cases.
- **Scope (0.20):** full — root cause traced through `kijun_sen` to the emitted signal; the
  do-not-touch list is enumerated.
- **Acceptance criteria (0.30):** full — every criterion is verifiable inside one `make pipeline`
  run; the unverifiable production-observation criterion was replaced with a deterministic test.
- **Technical context (0.20):** full — two existing in-repo guard precedents identified for the
  implementer to copy.
- **Residual uncertainty:** the upstream source of the zero-price candles is not identified by this
  ticket and is not in its scope. This ticket makes the consumer fail safely; finding why zero-price
  candles are produced at all belongs to `petrosa_k8s#1112`.

**Comment by yurisa2** [comment]
### MemPalace pre-flight report

**Dispatcher:** ticket-orchestrator
**Ticket:** PetroSa2/petrosa-bot-ta-analysis#320 — "[AUTOSRE] fix(bot-ta-analysis): Validation failed for ichimoku_cloud_momentum: risk parameters must be positive. SL: 0...."
**Affected service phrase:** "bot-ta-analysis ichimoku cloud momentum degenerate-data guards"

**Searches run**
- semantic search on title — unavailable
- kg query on `PetroSa2/petrosa-bot-ta-analysis#320` — unavailable
- tunnel discovery on service phrase — unavailable

**Key findings**
- MemPalace MCP tools are unavailable in this runtime; no institutional-memory results were available.

**Decision impact**
- Proceeding from the issue specification and repository state without MemPalace context.

**Repo-drift flag:** no
**MemPalace availability:** UNAVAILABLE (MemPalace tools are not exposed in this runtime)

**Comment by yurisa2** [comment]
### BMAD lifecycle: work started

Work on this ticket has started. Project board status: **In progress**.

**Comment by yurisa2** [comment]
### BMAD lifecycle: ready for review

Pull request: https://github.com/PetroSa2/petrosa-bot-ta-analysis/pull/324

Project board status: **In review**.

**Comment by yurisa2** [comment]
### BMAD lifecycle: review verdict — **Accept**

Self-review accepted: the Ichimoku strategy now rejects unusable closes and non-positive or non-finite stop-loss and take-profit values before signal construction. Regression coverage covers the 26-bar and 5-bar corruption cases plus both risk-guard branches; CI, security, lint, tests, and Codecov passed.

Project board status: **In review** (awaiting merge phase).

**Comment by yurisa2** [comment]
### BMAD lifecycle: merge complete

The change has been merged. PR: https://github.com/PetroSa2/petrosa-bot-ta-analysis/pull/324

Project board status: **Done**.

**Comment by yurisa2** [comment]
### BMAD lifecycle: finalization complete

PR #324 merged the Ichimoku strategy guards for unusable closes and non-positive or non-finite risk parameters. Regression coverage now exercises 26-bar and 5-bar zero-price corruption plus the risk-guard branches. CI, security, lint, tests, and Codecov passed; deployment verification was skipped because this is not the petrosa_k8s repository.
